### PR TITLE
Add pre-commit and github action checks

### PR DESCRIPTION
**Pull request by jojortz**
- add isort, autoflake, and mypy to pre-commit
- add github action to check dependencies, lint with flake8, and run unittests
- Fix unittests

### CODE REVIEW

**Comment by goldmermaid** [comment]
@jojortz per our discussion, can you comment the broken unit test for now and resubmit this pr.

**Review by goldmermaid — APPROVED** [review]

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,11 +1,35 @@
...
+    rev: v1.6.1
+    hooks:
+      - id: autoflake
+        args:
+          - "--in-place"
+          - "--expand-star-imports"
+          - "--remove-duplicate-keys"
+          - "--remove-unused-variables"
+          - "--remove-all-unused-imports"
+        exclude: "uniflow/__init__.py"
```
Comment (goldmermaid) [inline .pre-commit-config.yaml:35]: nit: add a new line

File: .github/workflows/python-app.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        poetry install --no-root # This will install the project dependencies defined in pyproject.toml
+    - name: Lint with flake8
+      run: |
+        # stop the build if there are Python syntax errors or undefined names
+        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+        flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+    - name: Test with unittest
+      run: |
+        poetry run python -m unittest discover
```
Comment (goldmermaid) [inline .github/workflows/python-app.yml:37]: per our discussion, let's also add isort and black here.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (jojortz) [reply .pre-commit-config.yaml]: Done!

File: .github/workflows/python-app.yml (hunk 2 above)
Comment (jojortz) [reply .github/workflows/python-app.yml:45]: Added!

**Pull request merged**